### PR TITLE
ci: Add pmem CI job

### LINKED ISSUE #3867: add setup for pmem job to build it

### PR DESCRIPTION
**Pull request by GabyCT**
This PR adds the possibility to build PMEM to avoid regressions or failures,
however, it will not run the test as this needs a special hardware.

Fixes #3867

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by GabyCT** [comment]
@devimc what else should be included from `make pmem` into this PR?

**Comment by GabyCT** [comment]
@devimc should this be running on baremetal?

**Review by devimc — APPROVED** [review]
thanks @GabyCT

**Comment by GabyCT** [comment]
@devimc I still got the error
```
10:06:04 NAMESPACE     NAME                                             READY   STATUS             RESTARTS   AGE
10:06:04 default       my-csi-kata-app                                  0/1     Pending            0          2m
10:06:04 kube-system   coredns-558bd4d5db-22zmh                         1/1     Running            0          10m
10:06:04 kube-system   coredns-558bd4d5db-jlcmn                         1/1     Running            0          10m
10:06:04 kube-system   etcd-ubuntu20-0473f0                             1/1     Running            0          10m
10:06:04 kube-system   kube-apiserver-ubuntu20-0473f0                   1/1     Running            0          10m
10:06:04 kube-system   kube-controller-manager-ubuntu20-0473f0          1/1     Running            0          10m
10:06:04 kube-system   kube-flannel-ds-vtprr                            1/1     Running            0          10m
10:06:04 kube-system   kube-proxy-ckg92                                 1/1     Running            0          10m
10:06:04 kube-system   kube-scheduler-ubuntu20-0473f0                   1/1     Running            0          10m
10:06:04 pmem-csi      pmem-csi-intel-com-controller-597989ff86-scxpj   1/1     Running            0          117s
10:06:04 pmem-csi      pmem-csi-intel-com-node-8dmfb                    1/3     CrashLoopBackOff   6          117s
10:06:04 pmem-csi      pmem-csi-operator-78598894d4-hdztd               1/1     Running            0          2m3s
```

File: integration/pmem/pmem_test.sh [hunk 1]
```diff
@@ -105,17 +95,33 @@ EOF
...
+	kubectl label node $(hostname) katacontainers.io/kata-runtime=true
+	popd
 }
 
 main() {
 	trap cleanup EXIT QUIT KILL
 	init
 	setup_pmem_csi
-	run_test
+	[[ CI_JOB != PMEM_BAREMETAL ]] && run_test
```
Comment (devimc) [inline integration/pmem/pmem_test.sh:124]: `$CI_JOB` missing `$`

Comment (devimc) [reply integration/pmem/pmem_test.sh:124]: and "PMEM_BAREMETAL"

**Comment by dborquez** [comment]
lgtm

**Review by chavafg — APPROVED** [review]

**Pull request merged**